pydantic_ai_slim/pydantic_ai/models/outlines.py

```diff
@@ -0,0 +1,514 @@
+        chat = Chat()
+        for message in messages:
+            if isinstance(message, ModelRequest):
+                for part in message.parts:
+                    if isinstance(part, SystemPromptPart):
+                        chat.add_system_message(part.content)
+                    elif isinstance(part, UserPromptPart):
+                        if isinstance(part.content, str):
+                            chat.add_user_message(part.content)
+                        elif isinstance(part.content, Sequence):
+                            if isinstance(part.content, str):
+                                chat.add_user_message(part.content)
+                            elif isinstance(part.content, Sequence):
+                                outlines_input: Sequence[str | Image] = []
+                                for item in part.content:
+                                    if isinstance(item, str):
+                                        outlines_input.append(item)
+                                    elif isinstance(item, BinaryImage):
+                                        image = PILImage.open(io.BytesIO(item.data))
+                                        image.format = item.media_type.split('/')[-1]
+                                        outlines_input.append(Image(image))
+                                    else:
+                                        raise UserError(
+                                            'Each element of the content sequence must be a string or a `BinaryImage`.'
+                                        )
+                                chat.add_user_message(outlines_input)
+                            else:
+                                assert_never(part.content)
+                        else:
+                            assert_never(part.content)
+                    elif isinstance(part, RetryPromptPart):
+                        chat.add_user_message(part.model_response())
+                    elif isinstance(part, ToolReturnPart):
+                        raise UserError('Tool calls are not supported for Outlines models yet.')
+                    else:
+                        assert_never(part)
+            elif isinstance(message, ModelResponse):
+                for part in message.parts:
+                    if isinstance(part, TextPart):
+                        chat.add_assistant_message(part.content)
+                    elif isinstance(part, ThinkingPart):
+                        # NOTE: We don't send ThinkingPart to the providers yet.
+                        pass
+                    elif isinstance(part, ToolCallPart | BuiltinToolCallPart | BuiltinToolReturnPart):
+                        raise UserError('Tool calls are not supported for Outlines models yet.')
+                    elif isinstance(part, FilePart):
+                        raise UserError('File parts are not supported for Outlines models yet.')
```

Comment:
Could we include assistant input images (https://ai.pydantic.dev/builtin-tools/#image-generation-tool) as well?